src/encode_append.rs

```diff
@@ -41,10 +48,37 @@ pub trait EncodeAppend {
 	/// // Add multiple element
 	/// <Vec<u32> as EncodeAppend>::append(encoded, &[700u32, 800u32, 10u32]).expect("Adds new elements");
 	/// ```
+	fn append_or_new<'a, EncodeLikeItem, I>(
+		self_encoded: Vec<u8>,
+		iter: I,
+	) -> Result<Vec<u8>, Error>
+	where
+		I: IntoIterator<Item = &'a EncodeLikeItem>,
+		EncodeLikeItem: 'a + EncodeLike<Self::Item>,
+		I::IntoIter: ExactSizeIterator;
+}
+
+impl<T: Encode> EncodeAppend for Vec<T> {
+	type Item = T;
+
 	fn append<'a, I: IntoIterator<Item=&'a Self::Item>>(
 		self_encoded: Vec<u8>,
 		iter: I,
-	) -> Result<Vec<u8>, Error> where Self::Item: 'a, I::IntoIter: ExactSizeIterator;
+	) -> Result<Vec<u8>, Error> where Self::Item: 'a, I::IntoIter: ExactSizeIterator {
+		append_or_new_vec_with_any_item(self_encoded, iter)
+	}
+
+	fn append_or_new<'a, EncodeLikeItem, I>(
+		self_encoded: Vec<u8>,
+		iter: I,
+	) -> Result<Vec<u8>, Error>
+	where
+		I: IntoIterator<Item = &'a EncodeLikeItem>,
```

Comment:
we do use a reference here because as we want to be able to append while not consuming value for instance `&([u32], u32)` implements `&EncodeLike<(Vec<u32>, u32)>` but not `EncodeLike<&(Vec<u32>, u32)>`